editor/plugins/material_editor_plugin.cpp

```diff
@@ -160,45 +162,68 @@ void MaterialEditor::edit(Ref<Material> p_material, const Ref<Environment> &p_en
 	}
 }
 
-void MaterialEditor::_on_light_1_switch_pressed() {
-	light1->set_visible(light_1_switch->is_pressed());
-}
+void MaterialEditor::_on_visibility_switch_pressed(const int &p_shape) {
+	switch ((Switch)p_shape) {
+		case LIGHT_1: {
+			light1->set_visible(light_1_switch->is_pressed());
+			EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_light1", light1->is_visible());
+		} break;
 
-void MaterialEditor::_on_light_2_switch_pressed() {
-	light2->set_visible(light_2_switch->is_pressed());
-}
+		case LIGHT_2: {
+			light2->set_visible(light_2_switch->is_pressed());
+			EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_light2", light2->is_visible());
+		} break;
 
-void MaterialEditor::_on_sphere_switch_pressed() {
-	sphere_instance->show();
-	box_instance->hide();
-	quad_instance->hide();
-	box_switch->set_pressed(false);
-	quad_switch->set_pressed(false);
-	_set_rotation(-15.0, 30.0);
-	_store_rotation_metadata();
-	EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_mesh", "sphere");
+		case FLOOR: {
+			bool is_visible = !floor_instance->is_visible();
+			floor_instance->set_visible(is_visible);
+			floor_switch->set_pressed(is_visible);
+			EditorSettings::get_singleton()->set_project_metadata("inspector_options", "material_preview_floor", is_visible);
+		} break;
+	}
 }
 
-void MaterialEditor::_on_box_switch_pressed() {
+void MaterialEditor::_on_shape_switch_pressed(const int &p_shape) {
 	sphere_instance->hide();
```

Comment:
It would be more efficient to do
```suggestion
	sphere_instance->set_visible(p_shape == SPHERE);
```
Avoids toggling things unnecessarily.

Reply:
Same with `set_pressed_no_signal()` 🙃